app/code/Magento/Newsletter/Model/RemoveSubscriberFromQueueLink.php

```diff
@@ -0,0 +1,44 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Newsletter\Model;
+
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\DB\Adapter\AdapterInterface;
+
+/**
+ * Responsible for removing subscriber from queue
+ */
+class RemoveSubscriberFromQueueLink
+{
+    /**
+     * @var AdapterInterface
+     */
+    private $connection;
+
+    /**
+     * @param ResourceConnection $resource
+     */
+    public function __construct(ResourceConnection $resource)
+    {
+        $this->connection = $resource->getConnection();
+    }
+
+    /**
+     * Removes subscriber from queue
+     *
+     * @param int $subscriberId
+     * @return void
+     */
+    public function execute(int $subscriberId): void
+    {
+        $this->connection->delete(
+            $this->connection->getTableName('newsletter_queue_link'),
```

Comment:
@engcom-Charlie I believe the table name should be retrieved from `ResourceConnection`

Reply:
I think is better to add normal resource model `QueueLink` that extends `AbstractDb` and use `getMainTable` in query.
@sivaschenko what do you think?

Reply:
I don't think we need to introduce a new class for a simple database operation.

Also, implementing granular commands is a better approach comparing to using resource models in terms of maintainability and extensibility.